src/lsm/grid.zig

```diff
@@ -358,16 +359,10 @@ pub fn GridType(comptime Storage: type) type {
                 .block_type = block_type,
             };
 
-            if (grid.read_recursion_guard) {
-                grid.read_recursion_queue.push(read);
-                return;
-            }
-
             grid.start_read(read);
         }
 
         fn start_read(grid: *Grid, read: *Grid.Read) void {
```

Comment:
Since the cache is accessed twice anyways (once which could queue the read to `read_cached`, and again in `tick()` to actually get the block), cache is assumed cheap to access so didn't want to over-complicate.